File: LeetCode/1-100/62. Unique Paths/src/Solution.java
class Solution {

    public int uniquePaths(int m, int n) {
        int num = m + n - 2;
        int den = 2;
        if (m - 1 > num / 2) {
            den = m;
            m = n;
            n = den;
            den = 2;
        }
        int ans = 1;
        while (num > n - 1 || den < m) {
            if (den < m && ans % den == 0) {
                ans /= den;
                den++;
            } else if (num > n - 1) {
                ans *= num;
                num--;
            }
        }
        return ans;
    }
}

File: LeetCode/1-100/62. Unique Paths/Solution.cpp
class Solution {
    int dp[101][101];

public:
    Solution() {
        for (int i = 0; i < 101; ++i) {
            dp[0][i] = 1;
        }
        long long temp;
        for (int i = 1; i < 101; ++i) {
            dp[i][0] = 1;
            for (int j = 1; j < 101; ++j) {
                temp = dp[i - 1][j];
                temp += dp[i][j - 1];
                dp[i][j] = temp < INT_MAX ? temp : 0;
            }
        }
    }

    int uniquePaths(int m, int n) {
        return dp[m - 1][n - 1];
    }
};